File: create_library.m
%% Create Tex File
clc;
close all;
clearvars;

fid     = fopen('PianoLibrary.tex','w');

%% Add generic structure
fprintf(fid, ['\\documentclass{article}' '\n']);
fprintf(fid, ['\\usepackage[utf8]{inputenc}' '\n']);
fprintf(fid, ['\\usepackage{pdfpages}' '\n']);
fprintf(fid, ['\\usepackage{hyperref}' '\n']);
fprintf(fid, ['\\title{Compilation of Piano Music}' '\n']);
fprintf(fid, ['\\author{Andrew Fisher}' '\n']);
fprintf(fid, ['\\begin{document}' '\n']);
fprintf(fid, ['\\maketitle' '\n']);
fprintf(fid, ['\\tableofcontents' '\n']);
fprintf(fid, ['\\clearpage' '\n']);

%% Loop over pdf files
% Read in pdf songfile names
songnames   = dir('PianoPDFs\*.pdf*');
for j=1:length(songnames)
    songtitle   = erase(songnames(j).name,'.pdf');
    fprintf(fid, ['\\phantomsection' '\n']);
    fprintf(fid, ['\\addcontentsline{toc}{section}{' songtitle '}' '\n']);
    fprintf(fid, ['\\includepdf[pages=-]{"PianoPDFs/' songtitle '"}' '\n']);
end

fprintf(fid, ['\\end{document}' '\n']);


%% Compile tex file into PDF music library
system('pdflatex -quiet PianoLibrary.tex');
fprintf('Compiled once\n');
system('pdflatex -quiet PianoLibrary.tex'); %compiling twice to ensure TOC exists
fprintf('Compiled twice\n');
fclose('all');

%% Send PDF to Test Email
send_email('Piano PDF','A searchable compilation of all piano pdfs','PianoLibrary.pdf')

File: create_library.py
# -*- coding: utf-8 -*-
"""
Created on Fri Jun 24 17:39:30 2022

@author: afish
"""
import os
import subprocess
folder = 'PianoPDFs'

# Create TEX document
with open('library.tex','w') as f:
    preamble = ['\\documentclass{article}',
                '\\usepackage[utf8]{inputenc}',
                '\\usepackage{pdfpages}',
                '\\usepackage{hyperref}',
                '\\title{PDF Library}',
                '\\author{Andrew Fisher}',
                '\\begin{document}',
                '\\maketitle',
                '\\tableofcontents',
                '\\clearpage','']
    f.write('\n'.join(preamble))
    
    for file in os.listdir(folder):
        if file.endswith('.pdf'):
            table_element = ['\\phantomsection',
                             '\\addcontentsline{toc}{section}{%s}'%file,
                             '\\includepdf[pages=-]{"%s/%s"}'%(folder,file),'']
            f.write('\n'.join(table_element))
    
    f.write('\\end{document}\n')
print('Created library.tex')

# Compile TEX document
cmd = subprocess.run('pdflatex -quiet library.tex'.split(' '), stderr=subprocess.PIPE, encoding='UTF=8')
cmd = subprocess.run('pdflatex -quiet library.tex'.split(' '), stderr=subprocess.PIPE, encoding='UTF=8')

# Remove extra files
extensions = ['aux','log','out','tex','toc']
for extension in extensions:
    os.remove('library.'+extension)


